develop/tutorials/articles/110-portlets/15-javascript/04-npm/08-npm-sdk.markdown

```diff
@@ -1,171 +1,208 @@
+          - Source maps support (https://issues.liferay.com/browse/LPS-75339)
+          - Variable aliases in require attribute of <aui:script> tag (https://issues.liferay.com/browse/LPS-75553)
+
+    Feature level 1
+      · Gradle: 2.5.0
+      · Gradle Node plugin: 2.3.0
+      · npm: 4.0.0
+      · Babel: 6.26.0
+      · Liferay npm bundler: 1.2.1
+      · Loader modules extender: 1.0.14
+      · AMD loader: 2.1.0
+      · Supported features:
+          - Basic functionality
+
++$$$
+
+**Note:** The tool requires access to a running @product@ instance to get the 
+AMD loader version and the Gogo console to get the 
+frontend-js-loader-modules-extender version. If you don't have a @product@ 
+instance available, the tools can't determine the version number for these 
+components and doesn't take them into account when calculating the feature 
+level. This can cause a feature level to appear higher than it actually is, 
+since those components are not accounted for.
+
+$$$
+
+As long your project has components with the minimum version numbers, you can 
+use the features listed in the feature level (plus all those inherited from the 
+previous feature level, of course). If you set a component to a higher version 
+number which is compatible according to semantic versioning, you are still in 
+the same feature level (or maybe a higher one if you increase version numbers 
+enough to reach a new level). This can lead to running an unsupported npm SDK 
+version that may be broken or produce unexpected results. For example, say that 
+feature level 2 is the maximum possible, and your component versions are 
+slightly higher than the minimum versions required (Loader modules extender has 
+increased from version 1.1.0 to 1.5.0):
+
+- Gradle: 2.5.0
+- Gradle Node plugin: 2.3.0
+- npm: 4.0.0
+- Babel: 6.26.0
+- Liferay npm bundler: 1.2.1
+- Loader modules extender: 1.5.0
+- AMD loader: 2.1.0
+
+You are still in feature level 2, as all version changes have been semantical 
+versioning compatible. If, however, you update to these version numbers:
+
+- Gradle: 2.5.0
+- Gradle Node plugin: 2.3.0
+- npm: 4.0.0
+- Babel: 6.26.0
+- Liferay npm bundler: 1.5.2
+- Loader modules extender: 2.1.0
+- AMD loader: 2.1.0
+
+You are no longer in feature level 2, as you have introduced the possibility of 
+a breaking change when updating Loader Modules Extender from 1.1.0 to 2.1.0. In 
+this case, you would be in an indeterminate feature level (pre-3 for example) 
+that has not been tested. We recommend that you use component versions that are 
+semantical version compatible with the component versions listed in the feature 
+level you wish to use. 
+
++$$$
+
+<!-- `liferay-npm-build-tools` version only shows up in your level 8 example in the README. Running 
+the tool does not list this component. Is it suppose to be listed, or has this 
+changed?  -->
+
+**Note:** The liferay-npm-build-tools version specifies the version for the 
```

Comment:
The "problem" is that `liferay-npm-build-tools` is the container for several babel and bundler plugins, so there's nothing like "using version x.y.z of liferay-npm-build-tools in a project" because you may be using different versions of those said plugins.

What I was probably referring to is that you are using all babel plugins and the like in the same version number x.y.z. That would be the only case, where you could say that you are using "version x.y.z of liferay-npm-build-tools": because all the subcomponents living in `liferay-npm-build-tools` are effectively version-aligned.

Reply:
Ok. I'll just update the note then to say that `liferay-npm-build-tools` refers to the container for those components, but not an actual component itself. The main take away is the synced release cycle and how that affects subcomponents.

Reply:
@izaera I'm referring to your note here: https://www.npmjs.com/package/liferay-npm-sdk-assistant
>And that it specifies the following minimum versions for each component (version numbers invented too):

>gradle: 12.0
gradle plugin for node: 17.0.0
npm: 8.0
babel: 21.0
liferay-npm-build-tools: 27.0.0
frontend-js-loader-modules-extender: 15.0.0
liferay-amd-loader: 8.0.0

>Note that the version of liferay-npm-build-tools specifies the version for the liferay-npm-bundler tool, the plugins for Babel, and the plugins for liferay-npm-bundler.

>In fact, all these components are always released together to make version management easier. That means that some subcomponent may have different version numbers for the same artifact, but we prefer such redundancy to having disparate version numbers for each subcomponent.

So, you're saying that `liferay-npm-build-tools` is just used in the example as a fake component?